apps/hash-api/src/graphql/resolvers/knowledge/page/page-contents.ts

```diff
@@ -1,35 +1,30 @@
 import { Entity } from "@local/hash-subgraph";
-import { ApolloError } from "apollo-server-errors";
 
-import {
-  getPageBlocks,
-  getPageById,
-} from "../../../../graph/knowledge/system-types/page";
+import { getPageBlocks } from "../../../../graph/knowledge/system-types/page";
 import { ResolverFn } from "../../../api-types.gen";
 import { LoggedInGraphQLContext } from "../../../context";
 import { dataSourcesToImpureGraphContext } from "../../util";
-import { mapBlockToGQL, UnresolvedPageGQL } from "../graphql-mapping";
+import {
+  mapBlockToGQL,
+  mapEntityToGQL,
+  UnresolvedBlockGQL,
+  UnresolvedPageGQL,
+} from "../graphql-mapping";
 
 export const pageContents: ResolverFn<
-  Promise<Entity[]>,
+  Promise<{ linkEntity: Entity; rightEntity: UnresolvedBlockGQL }[]>,
   UnresolvedPageGQL,
   LoggedInGraphQLContext,
   {}
-> = async ({ metadata }, _, { dataSources }) => {
+> = async (page, _, { dataSources }) => {
   const context = dataSourcesToImpureGraphContext(dataSources);
 
-  const entityId = metadata.recordId.entityId;
-  const page = await getPageById(context, { entityId });
-
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- @todo improve logic or types to remove this comment
-  if (!page) {
-    throw new ApolloError(
-      `Page with entityId ${entityId} not found`,
-      "NOT_FOUND",
-    );
-  }
-
-  const blocks = await getPageBlocks(context, { page });
```

Comment:
This is the unnecessary fetching that prompted the change to the signature of `getEntityOutgoingLinks` – we were fetching the page again to be able to pass it into `getPageBlocks`, which only really needs the id (which we already have).